### DISCUSSION TITLE
How to Disable sending emails when a package is published ?

### DISCUSSION THREAD

**Discussion by romain130492** [npm/feedback#424] [General]
Hey how to Disable sending emails when a package is published ?

I already saw many post about this same question  : https://github.com/npm/registry-issue-archive/issues/284

Thanks

**Comment by MylesBorins** [comment]
The intent behind the published email notifications is to ensure you will be notified if your account is ever compromised and your credentials are used to publish packages you did not intend to publish. We are discussed features that will allow you to apply preferences to how often you receive these emails, but this work is not currently on the roadmap.

In the meantime, you may choose to filter them using your email provider's settings.

**Reply by farfromrefug** [reply]
I want that feature too. And it is not a question of filtering emails. Think about the energy / disk space used to send all those emails? We all have a role to play in reducing global energy consumption. 
Please this is not an `good to have` feature, it is a `must have feature`

**Reply by NeonWizard** [reply]
Another thing I'd like to suggest is the difference between automatic publishes and manual publishes. Perhaps add a feature to only get emails when someone manually pushes a new version to the registry, as opposed to when using an `Automation` type token to push to NPM via whatever form of automation? The product I work on publishes anywhere from 1-5 patch versions of a library every day via our CI/CD, leading to a lot of mailbox spam. I don't want to just filter as spam in my email provider, because I do actually want to know when a developer manually publishes a release.

**Comment by fabswt** [comment]
Too many emails means that the day an email matters it'll be either lost in the noise or hidden by a filter.

I joined yesterday, got a paid account, and got 30 emails already. They're private test packages and it makes absolute no sense for me to get notifications for them.

**Comment by fabswt** [comment]
[Dozens of requests for the feature in this archived thread](https://github.com/npm/registry-issue-archive/issues/284).

**Comment by f3oall** [comment]
@MylesBorins any progress on this issue? It seems really annoying, and I bet a lot of people would like to have notification preferences. Actually, it's pretty weird that suck a significant application doesn't have it.

So it would be really nice if you could add this feature. Thanks!

**Comment by softmarshmallow** [comment]
I get 20 emails every time i publish my lerna repo.
Would be nice to bundle to one notification for same organization with time debounce?

**Comment by planetf1** [comment]
Finding this very frustrating too. I will setup a mail rule, but there's just too much noise currently.

**Comment by mathenry1** [comment]
I'm getting a lot mails from spam people with contact details of my customers
My customers orders my products in credit from me but some people hacked their account and send mails to me and keep ordering So I want to stop and refund my products, how can I stop them?
I have a website based on [Buy Painkiller](https://onlinemedsstore.uk/) and Pain Pills from ONLINE MEDS STORE

**Comment by azerum** [comment]
Created a separate mail for NPM. Also helps with the fact that NPM makes your email public. I've read about it in https://github.com/npm/policies/issues/58, and from last comments I understood that the intent of exposing your email is to allow other developers to easily contact you, but it's hard to keep track of mail when there are so many publish notifications anyway

**Comment by azu** [comment]
Especilly, This behavior send many mails to monorepo user at same time.
I've used Gmail filter + Google App Scripts(GAS).

| Before | After|
| ----- | ---- |
| ![image](https://user-images.githubusercontent.com/19714/212238864-defb53d4-3b1e-4022-a02d-c28a2bc4df1b.png) | ![image](https://user-images.githubusercontent.com/19714/212239101-149d87e2-6d44-4f38-beb6-7a5e646098c8.png) |

This GAS create a single digest mail for "Successfully published" mails.

- [azu/gmail-npm-publish-digest: Google App Scripts create digest mail of npm's &quot;Successfully published&quot; notification from your Gmail read mails.](https://github.com/azu/gmail-npm-publish-digest)

If npm provide digest mail like GitHub's dependabot, It will reduce pain.

<img width="835" alt="image" src="https://user-images.githubusercontent.com/19714/212239572-b0d15c90-ccb3-4aaf-a579-8aa3d19befb7.png">

**Comment by qq253498229** [comment]
Are there any rules currently configurable? 
What if tag is a beta version that doesn't send emails?

**Comment by aedart** [comment]
I too would really like an option to either disable or "bundle"  publish success or failure emails. Currently they provide little value at all, especially when publishing with multiple packages at once (_e.g. mono-repository_).

**Comment by matyasf** [comment]
Any updates? I'm managing a large monorepo with 100+ packages, so I get 100+ emails on a release. they all are deleted no immediately thanks to a gmail filter.

**Comment by poolsimon** [comment]
Can you pleease add an option to disable those mails. we publish on every push

**Comment by shgysk8zer0** [comment]
I get so many emails from npm that it makes my email effectively useless (could probably set up a filter).

This has been demanded for at least 5 years now and I'm about to block npm emails as spam. I really don't need 30 emails a day about things that I did! Stop it!!!

**Comment by TGChrisRArendt** [comment]
Looking forward to this being implemented. Should be pretty easy to set up as it's just a boolean toggle.

**Comment by meszaros-lajos-gyorgy** [comment]
Let me join the bandwagon of adding also my name to the list of people who would love to see any improvement on this subject.

**Comment by hwebb** [comment]
There could be a whitelist of IP's where we shouldn't receive any notifications, like that your argument of compromised account is solved. Or you can have it automatically where when you have the preference of no notification, it will detect any new IP and only send 2 notifications and stop sending when you see we didn't take action on that IP.
Because right now I just created a filter to auto-delete emails, which defeats the purpose you've intended.

**Comment by robertherber** [comment]
This would clean up my inbox :)

**Comment by MatthD** [comment]
re-up this, very annoying for a service like nom, I am publishing a package and testing CI for about 10 version of os each .. terrible!

**Comment by csukuangfj** [comment]
Any update? I'd like also to disable email notifications.

**Comment by ClydeDz** [comment]
+1 - would be nice to toggle this feature if and when needed. Getting too many emails for every version published.

**Comment by anasouardini** [comment]
And yet my mail box is still full of these annoying emails. From all NPM quircks, this is the most annoying-easy-to-fix-and-yet-still-not-fixed one so far. If anything, this only discourages me from sharing my personal dev tool with other devs.

It's disappointing given that NPM is around for more than 10 years:

![image](https://github.com/npm/feedback/assets/114059811/22ddd73b-6566-4ead-ac22-c3db1898a07f)

**Comment by colin969** [comment]
NPM isn't made to work for the developer, it's made to beat the developer with a stick into their way of thinking. The failure of a lot of projects unfortunately.

The problem comes from NPM being the only one you can realistically use, so they'll never change.

They claim forcing emails means people will spot bad activity, but it actually means people have to mark legitimate emails as spam, so they'll never see it.

**Comment by georg-ikegps** [comment]
It would be great to be able to specify the address to which the notifications go.

**Reply by anasouardini** [reply]
The only way you can deal with this is to have two separate addresses: one for spam and another for serious matter.

**Comment by anasouardini** [comment]
Although [JSR](https://jsr.io/) is not meant to replace `npm`, I'm still waiting for the day they're capable/ready to do so, and help developers share code without the discomfort that comes with `npm`.

**Comment by ts-web** [comment]
So @MylesBorins said...
1) the emails are good and you should read them
2) you can disable them with your email provider

Seems a bit odd to recommend that. But ok I will do that. Sadly, mobile email clients don't expose fine-grained filtering features. So what I can do is block _all_ mail from npm. That's not a good situation for nom, but 🤷 

Use the rocket emoji if you're doing this too.
